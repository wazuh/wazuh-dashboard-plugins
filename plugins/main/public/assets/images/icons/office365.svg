<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 66 81" fill="#fff" fill-rule="evenodd" stroke="#000" stroke-linecap="round" stroke-linejoin="round"><use xlink:href="#A" x=".5" y=".5"/><symbol id="A" overflow="visible"><path d="M65 73.333h0V7.027L41.811 0 .176 16.216H0v47.928l14.054-5.766V19.459l28.108-6.847V70.09L0 64.144 41.811 80h0L65 73.514v-.18z" fill="#006BB4" stroke="none"/></symbol></svg>