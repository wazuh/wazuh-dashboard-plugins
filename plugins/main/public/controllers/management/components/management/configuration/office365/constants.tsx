/*
 * Wazuh app - Constants of Office365 configuration
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import { DOC_LINKS } from '../../../../../../../common/doc-links';

export const OFFICE_365 = 'office365';
export const WMODULES_WMODULES = 'wmodules-wmodules';
export const HELP_LINKS = [
  {
    text: 'Monitoring Office 365',
    href: DOC_LINKS.CLOUD_SECURITY.OFFICE365.INDEX,
  },
  {
    text: 'Configuration options for the module',
    href: DOC_LINKS.USER_MANUAL.REFERENCE.OSSEC_CONF.OFFICE365_MODULE,
  },
];
