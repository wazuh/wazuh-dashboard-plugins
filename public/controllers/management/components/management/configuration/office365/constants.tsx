/*
 * Wazuh app - Constants of Office365 configuration
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import { webDocumentationLink } from "../../../../../../../common/services/web_documentation";

export const OFFICE_365 = 'office365';
export const WMODULES_WMODULES = 'wmodules-wmodules';
export const HELP_LINKS = [
  {
    text: 'Using Wazuh to monitor Office 365',
    href: webDocumentationLink('office365/index.html'),
  },
  {
    text: 'Configuration options for the module',
    href: webDocumentationLink('user-manual/reference/ossec-conf/office365-module.html'),
  },
];
