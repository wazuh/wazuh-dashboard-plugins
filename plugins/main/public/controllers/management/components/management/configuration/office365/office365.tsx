/*
 * Wazuh app - React component for show configuration of Office 365 module.
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React, { useEffect, useMemo } from 'react';
import WzNoConfig from '../util-components/no-config';
import withWzConfig from '../util-hocs/wz-config';
import { wodleBuilder } from '../utils/builders';
import { compose } from 'redux';
import WzTabSelector, { WzTabSelectorTab } from '../util-components/tab-selector';
import { isString } from '../utils/utils';
import { withGuard } from '../../../../../../components/common/hocs';
import { HELP_LINKS, OFFICE_365, WMODULES_WMODULES } from './constants';
import { GeneralTab } from './components/general-tab/general-tab';
import { ApiAuthTab } from './components/api-auth-tab/api-auth-tab';
import { SubscriptionTab } from './components/SubscriptionTab/SubscriptionTab';

interface IWzConfigOffice365 {
  currentConfig: {};
  agent: { id: string | number };
  updateBadge: () => void;
  updateConfigurationSection: () => void;
}
const sections = [{ component: 'wmodules', configuration: 'wmodules' }];

export const WzConfigurationOffice365: React.FunctionComponent<IWzConfigOffice365> = withWzConfig(
  sections
)(({ currentConfig, updateBadge, ...props }) => {
  const wodleConfiguration = useMemo(() => wodleBuilder(currentConfig, OFFICE_365), [
    currentConfig,
  ]);

  useEffect(() => {
    updateBadge(
      currentConfig &&
        wodleConfiguration &&
        wodleConfiguration[OFFICE_365] &&
        wodleConfiguration[OFFICE_365].enabled === 'yes'
    );
  }, [currentConfig]);

  return (
    <WzTabSelector>
      <WzTabSelectorTab label="General">
        <GeneralTabWrapped
          wodleConfiguration={wodleConfiguration}
          currentConfig={currentConfig}
          {...props}
        />
      </WzTabSelectorTab>
      <WzTabSelectorTab label="Credentials">
        <ApiAuthTabWrapped
          wodleConfiguration={wodleConfiguration}
          currentConfig={currentConfig}
          {...props}
        />
      </WzTabSelectorTab>
      <WzTabSelectorTab label="Subscriptions">
        <SubscriptionTabWrapped
          wodleConfiguration={wodleConfiguration}
          currentConfig={currentConfig}
          {...props}
        />
      </WzTabSelectorTab>
    </WzTabSelector>
  );
});

const tabWrapper = compose(
  withGuard(
    ({ currentConfig }) =>
      currentConfig[WMODULES_WMODULES] && isString(currentConfig[WMODULES_WMODULES]),
    ({ currentConfig }) => <WzNoConfig error={currentConfig[WMODULES_WMODULES]} help={HELP_LINKS} />
  ),
  withGuard(
    ({ wodleConfiguration }) => !wodleConfiguration[OFFICE_365],
    (props) => <WzNoConfig error="not-present" help={HELP_LINKS} />
  )
);

const GeneralTabWrapped = tabWrapper(GeneralTab);
const ApiAuthTabWrapped = tabWrapper(ApiAuthTab);
const SubscriptionTabWrapped = tabWrapper(SubscriptionTab);
