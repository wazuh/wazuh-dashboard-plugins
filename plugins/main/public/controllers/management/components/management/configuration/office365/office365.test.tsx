/*
 * Wazuh app - React Test component WzConfigurationOffice365
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { WzConfigurationOffice365 } from './office365';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

jest.mock('../../../../../../kibana-services', () => ({
  getUiSettings: () => ({
    get: (uiSetting: string) => {
      if (uiSetting === 'theme:darkMode') {
        return false;
      }
    },
  }),
}));

const mockStore = configureMockStore();
const store = mockStore({});

// the jest.mock of @osd/monaco is added due to a problem transcribing the files to run the tests.
// https://github.com/wazuh/wazuh-dashboard-plugins/pull/6921#issuecomment-2298289550

jest.mock('@osd/monaco', () => ({
  monaco: {},
}));

describe('WzConfigurationOffice365 component', () => {
  it('renders correctly to match the snapshot', () => {
    const agent = { id: '000' };

    const wrapper = shallow(
      <Provider store={store}>
        <WzConfigurationOffice365
          currentConfig={'master-test-node'}
          agent={agent}
          updateBadge={() => {}}
          updateConfigurationSection={() => {}}
        />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
