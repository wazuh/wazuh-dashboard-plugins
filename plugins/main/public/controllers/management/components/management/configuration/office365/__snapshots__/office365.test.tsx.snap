// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WzConfigurationOffice365 component renders correctly to match the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(WithLoading(Component))
    agent={
      Object {
        "id": "000",
      }
    }
    currentConfig="master-test-node"
    updateBadge={[Function]}
    updateConfigurationSection={[Function]}
  />
</ContextProvider>
`;
